corehq/apps/app_manager/util.py

```diff
@@ -409,7 +409,7 @@ def get_casedb_schema(form):
     This lists all case types and their properties for the given app.
     """
     app = form.get_app()
-    base_case_type = form.get_module().case_type
+    base_case_type = form.get_module().case_type if form.requires_case() else None
```

Comment:
> > This looks like a good change, but wondering if it will hide case properties from people who were previously seeing/using them?
>
> I believe the strict answer to this question is "no."

In that case, I approve this PR. I'm still in favor of continuing to evaluate whether #15798 can be further optimized or should be reverted.